src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -74,6 +72,16 @@ private static void PingResultValidator(PingReply pingReply, IPAddress[] localIp
             Assert.Contains(pingReply.Address, localIpAddresses); ///, "Reply address {pingReply.Address} is not expected local address.");
         }
 
+        private static byte[] GetPingPayload(AddressFamily addressFamily)
+            // On Unix, Non-root processes cannot send arbitrary data in the ping packet payload
+            => Capability.CanUseRawSockets(addressFamily) || PlatformDetection.IsOSXLike
+                ? TestSettings.PayloadAsBytes
+                : Array.Empty<byte>();
+
+        public static bool DoesNotUsePingUtility => !UsesPingUtility;
+
+        public static bool UsesPingUtility => OperatingSystem.IsLinux() && !(Capability.CanUseRawSockets(TestSettings.GetLocalIPAddress().AddressFamily) || PlatformDetection.IsOSXLike);
```

Comment:
The `||  PlatformDetection.IsOSXLike` looks wrong. IsOSXLike includes IsOSX where we use unprivileged ICMP raw socket AFAIK.

Reply:
Removed

Reply:
Does it need `|| OperatingSystem.IsAndroid()` as well ? `System.Net.NetworkInformation.Tests.PingTest.SendPingWithIPAddressAndBigSize` test now fails with PNSE on Android arm/arm64 leg, see https://github.com/dotnet/runtime/issues/64798